homeworks/03-data-structure/implementations/binary-tree.ts

```diff
@@ -0,0 +1,79 @@
+import {IBinaryTree} from "../models/binary-tree";
+import {TraverseType} from "../models/traverse-type";
+import {TreeNode} from "../models/tree-node";
+
+export class BinaryTree<T> implements IBinaryTree<T> {
+
+    constructor(protected root: TreeNode<T>) {
+    }
+
+    setTree(tree: TreeNode<T>): this {
+        this.root = tree;
+        return this;
+    }
+
+    traverse(traverseType: TraverseType): T[] {
+        if (traverseType === TraverseType.BFS) return this.bfs([this.root], []);
+
+        return this.dfs(this.root, traverseType);
+    }
+
+    getColumn(columnOrder: number): T[] {
+        return this.combineByLevel(columnOrder, 0, this.root);
+    }
+
+    insert(value: T, current?: TreeNode<T>): BinaryTree<T> {
```

Comment:
I would say you should not have a `current` argument here. Expected that this method should work with the existing tree. You create an instance and work with them. If you want to have some tree logic without the current tree, it should be `static` methods, but not in this case from my point of view.